### DISCUSSION TITLE
Impersonation functionality over Realms

### DISCUSSION THREAD

**Discussion by EwoutvdWielen** [keycloak/keycloak#43735] [Q&A]
### Description

We'd like to implement the impersonation-functionality of Keycloak so we can manage the environment of our customers. But I think you cannot impersonate users from other realms, am I right? Is there some solution or workaround for this use case?

**Comment by ericmorand** [comment]
I'd like to know too. More generally, how does impersonation works. There is a way, from the administration console, to impersonate another user. And there is a role, named "impersonation", created by default in the "master-realm" client, and a endpoint to actually impersonate another user...but there is no explanation about how all these things work.

What is the purpose of the "impersonation" role?

We created a role named "impersonation" in our client, and assigned this role to a user. When executing the impersonation endtpoint with the user access token, we get a `403 Forbidden`.

Now, if we assign to the user the "master-realm" client role "impersonation", it works!

It seems like the "master-realm" client has some privileges that _can't_ be given to any other client.

This makes no sense.
